### PR TITLE
Add artifact upload

### PR DESCRIPTION
**Pull request by Rodrigo-Tenorio**
Extend #147 by saving the output as artifacts (#149 )

### CODE REVIEW

**Comment by dbkeitel** [comment]
The test failures are probably due to git.ligo.org maintenance (fails to download ephemerides).

File: .github/workflows/run_examples.yml [hunk 1]
```diff
@@ -37,3 +37,12 @@ jobs:
         # make sure to test *installed* version of pyFstat
         pip install -e $GITHUB_WORKSPACE
         (cd .. && python $GITHUB_WORKSPACE/examples/run_all_examples.py)
+        examples/*/Pyfstat_example_data
+    - name: Save output files as artifacts (*.sft files excluded!)
+      uses: actions/upload-artifact@v2
+      with:
+        path: |
+          $GITHUB_WORKSPACE/examples/*/Pyfstat_example_data/*
+          !$GITHUBWORKSPACE/examples/*/Pyfstat_example_data/*/*.sft
```
Comment (dbkeitel) [inline .github/workflows/run_examples.yml:46]: Going by [this failed run](https://github.com/PyFstat/PyFstat/runs/1087472020?check_suite_focus=true), by how the paths are built locally, and the `..` in L39 above (of which I don't really remember why it's there :wink: ): I guess these should just be just `$GITHUB_WORKSPACE/Pyfstat_example_data/*`

Comment (Rodrigo-Tenorio) [reply .github/workflows/run_examples.yml]: Just did an `ls $GITHUB_WORKSPACE`:
```
 CHANGELOG.md
Dockerfile
LICENSE
MANIFEST.in
README.md
ephem
examples
pyfstat
requirements.txt
setup.cfg
setup.py
tests.py
versioneer.py
```

I'll try not to use wild cars and upload folders.

**Comment by Rodrigo-Tenorio** [comment]
```
Executing 1 script(s) in example directory /home/runner/work/PyFstat/PyFstat/examples/binary_examples...
Running:  /home/runner/work/PyFstat/PyFstat/examples/binary_examples/PyFstat_example_semi_coherent_binary_search_using_MCMC.py
Successfully ran: /home/runner/work/PyFstat/PyFstat/examples/binary_examples/PyFstat_example_semi_coherent_binary_search_using_MCMC.py
Base dir: /home/runner/work/PyFstat/PyFstat/examples
```

Running `ls $GITHUB_WORKSPACE`
```
drwxr-xr-x 10 runner docker  4096 Sep  9 14:58 .
drwxr-xr-x  3 runner docker  4096 Sep  9 14:57 ..
drwxr-xr-x  8 runner docker  4096 Sep  9 14:58 .git
-rw-r--r--  1 runner docker    33 Sep  9 14:57 .gitattributes
drwxr-xr-x  3 runner docker  4096 Sep  9 14:57 .github
-rw-r--r--  1 runner docker    10 Sep  9 14:57 .gitignore
-rw-r--r--  1 runner docker   944 Sep  9 14:57 .zenodo.json
-rw-r--r--  1 runner docker  5230 Sep  9 14:57 CHANGELOG.md
-rw-r--r--  1 runner docker   777 Sep  9 14:57 Dockerfile
-rw-r--r--  1 runner docker  1081 Sep  9 14:57 LICENSE
-rw-r--r--  1 runner docker   104 Sep  9 14:57 MANIFEST.in
drwxr-xr-x  2 runner docker  4096 Sep  9 14:58 PyFstat.egg-info
drwxr-xr-x  3 runner docker  4096 Sep  9 14:58 PyFstat_example_data
-rw-r--r--  1 runner docker 13234 Sep  9 14:57 README.md
drwxr-xr-x  2 runner docker  4096 Sep  9 14:58 __pycache__
drwxr-xr-x  2 runner docker  4096 Sep  9 14:58 ephem
drwxr-xr-x 10 runner docker  4096 Sep  9 14:57 examples
```
and `ls $GITHUB_WORKSPACE/examples`
```
 drwxr-xr-x 10 runner docker 4096 Sep  9 14:57 .
drwxr-xr-x 10 runner docker 4096 Sep  9 14:58 ..
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 binary_examples
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 followup_examples
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 glitch_examples
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 grid_examples
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 mcmc_examples
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 mcmc_vs_grid_simple_example
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 other_examples
-rw-r--r--  1 runner docker 2873 Sep  9 14:57 run_all_examples.py
drwxr-xr-x  2 runner docker 4096 Sep  9 14:57 transient_examples
```
drwxr-xr-x  4 runner docker  4096 Sep  9 14:58 pyfstat

**Comment by dbkeitel** [comment]
The crucial bit would be: `ls $GITHUB_WORKSPACE/PyFstat_example_data`

**Comment by Rodrigo-Tenorio** [comment]
Relative paths work wonders (and [this issue](https://github.com/actions/upload-artifact/issues/110) says so).

**Comment by Rodrigo-Tenorio** [comment]
@dbkeitel I think this is it (let's wait for the completion of [the full action](https://github.com/PyFstat/PyFstat/runs/1092332802?check_suite_focus=true))

**Comment by Rodrigo-Tenorio** [comment]
Done! [Here is the artifact](https://github.com/PyFstat/PyFstat/suites/1166825119/artifacts/16968476)

**Review by dbkeitel — APPROVED** [review]
I can't say I understand **why** this is the solution (would really have expected that `$GITHUB_WORKSPACE` needs to always be used for all paths), but it seems to work great. Thanks a lot!

**Pull request merged**